pkg/ruler/ruler.go

```diff
@@ -171,36 +172,40 @@ func NewRuler(cfg Config, engine *promql.Engine, queryable storage.Queryable, d
 		workers[i] = newWorker(ruler)
 	}
 
-	ruler.workers = workers
-	go ruler.run()
-
-	return ruler, nil
-}
-
-// Run the ruler.
-func (r *Ruler) run() {
-	go r.scheduler.Run()
-	for _, w := range r.workers {
-		go w.Run()
+	go ruler.scheduler.Run()
+	for i := range workers {
```

Comment:
Could this and the loop above be simplified to something like:
```
ruler.workers.Add(cfg.NumWorkers)
for i := 0; i < cfg.NumWorkers; i++ {
    go func() {
        newWorker(ruler).Run()
        ruler.workers.Done()
    }()
}
```
It doesn't look like we actually need the slice of workers anymore.

Reply:
True, I updated and removed the slice entirely

Reply:
:+1: